Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityInputManager.cs

```diff
@@ -1702,5 +1743,156 @@ public void RaiseDictationError(IMixedRealityInputSource source, string dictatio
         #endregion Dictation Events
 
         #endregion Input Events
+
+        #region Rules
+
+        private MixedRealityInputAction ProcessRules(MixedRealityInputAction inputAction, bool criteria)
+        {
+            if (CurrentInputActionRulesProfile != null)
+            {
+                var digitalRules = CurrentInputActionRulesProfile.InputActionRulesDigital;
+
+                for (int i = 0; i < digitalRules.Length; i++)
+                {
+                    if (digitalRules[i].BaseAction == inputAction && digitalRules[i].Criteria == criteria)
+                    {
+                        if (digitalRules[i].RuleAction == inputAction)
+                        {
+                            Debug.LogError("Input Action Rule cannot be the same as the rule's Base Action!");
+                            return inputAction;
+                        }
+
+                        return digitalRules[i].RuleAction;
+                    }
+                }
+            }
+
+            return inputAction;
+        }
+
+        private MixedRealityInputAction ProcessRules(MixedRealityInputAction inputAction, float criteria)
+        {
+            if (CurrentInputActionRulesProfile != null)
+            {
+                var digitalRules = CurrentInputActionRulesProfile.InputActionRulesSingleAxis;
+
+                for (int i = 0; i < digitalRules.Length; i++)
+                {
+                    if (digitalRules[i].BaseAction == inputAction && digitalRules[i].Criteria.Equals(criteria))
+                    {
+                        if (digitalRules[i].RuleAction == inputAction)
```

Comment:
Don't think we need this check.  It could be possible to configure a rule with the same action.
It's up to the user at the end of the day how they configure it.

Maybe only log a warning, just to inform them.   But better that is done in the inspector rather than here.

Reply:
Well, if we don't check it here and someone sets it up grammatically it's an infinite loop

Reply:
You shouldn't be raising an action you're consuming.